src/interfaces.ts

```diff
@@ -84,25 +84,31 @@ export interface ICoreKit {
   /**
    * Login into the SDK in an implicit flow and initialize all relevant components.
    * @param loginParams - Parameters for Implicit Login.
-   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. Optional for new users, mandatory for existing users, if not provided we will try to fetch it from local storage.
-   * @returns A Web3 provider if we are not in redirect mode.
    */
-  login(loginParams: LoginParams, factorKey?: BN): Promise<SafeEventEmitterProvider | null>;
+  loginWithOauth(loginParams: OauthLoginParams): Promise<void>;
 
   /**
    * Login into the SDK using ID Token based login and initialize all relevant components.
    * @param idTokenLoginParams - Parameters with ID Token based Login.
-   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. Optional for new users, mandatory for existing users, if not provided we will try to fetch it from local storage.
-   * @returns A Web3 provider
    */
-  loginWithIdToken(idTokenLoginParams: IdTokenLoginParams, factorKey?: BN): Promise<SafeEventEmitterProvider | null>;
+  login(idTokenLoginParams: IdTokenLoginParams): Promise<void>;
 
   /**
    * Handle redirect result after login.
-   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. Optional for new users, mandatory for existing users, if not provided we will try to fetch it from local storage.
-   * @returns A Web3 provider.
    */
-  handleRedirectResult(factorKey?: BN): Promise<SafeEventEmitterProvider>;
+  handleRedirectResult(): Promise<void>;
+
+  /**
+   * Second step for login where the user inputs their factor key.
+   * @param factorKey: A BN used for encrypting your Device/ Recovery TSS Key Share. You can generate it using `generateFactorKey()` function or use an existing one.
+   */
+  inputFactorKey(factorKey: BN): Promise<void>;
```

Comment:
`inputFactorKey` doesn't convey much about what this function is doing. A more descriptive function name could be something like below?
```suggestion
  finalizeLogin(factorKey: BN): Promise<void>;
```
And then maybe call the other login functions `initLogin...`.

Also, would be great if it could briefly be outlined here what was the reasoning behind changing from providing the factor key via the login function to providing the factor key via a separate function, as it is done now.